src/pages/[...arbitrary].tsx

```diff
@@ -107,6 +109,12 @@ export const getStaticProps: GetStaticProps<StaticProps> = async (context) => {
 };
 
 const IframePage: NextPageWithLayout = (props: InferGetStaticPropsType<typeof getStaticProps>) => {
+  const setComponentSrc = useCurrentComponentStore((store) => store.setSrc);
+
+  useEffect(() => {
+    setComponentSrc(null);
+  }, [setComponentSrc]);
```

Comment:
since we do this same logic in a number of places, it might be nice to see this as a custom `useClearCurrentComponent()` hook

fine either way, feel free to merge as is